sds_data_manager/constructs/ialirt_processing_construct.py

```diff
@@ -189,8 +204,90 @@ def add_compute_resources(self):
             desired_count=1,
         )
 
+    def create_autoscaling_event_rule(
+        self,
+        assign_eip_lambda: lambda_alpha_.PythonFunction,
+        auto_scaling_group: autoscaling.AutoScalingGroup,
+    ) -> None:
+        """Create Rules to trigger Lambda on Auto Scaling Group instance launch."""
+        deploy_rule = events.Rule(
+            self,
+            "AssignEipOnEc2InstanceLaunch",
+            rule_name="assign-eip-ec2-instance-launch",
+            event_pattern=events.EventPattern(
+                source=["aws.ec2"],
+                detail_type=["EC2 Instance State-change Notification"],
+                detail={
+                    "state": ["running"],
+                },
+            ),
+        )
+        asg_lifecycle_rule = events.Rule(
+            self,
+            "AssignEipOnInstanceLaunch",
+            rule_name="assign-eip-instance-launch",
+            event_pattern=events.EventPattern(
+                source=["aws.autoscaling"],
+                detail_type=["EC2 Instance-launch Lifecycle Action"],
+                detail={
+                    "AutoScalingGroupName": [auto_scaling_group.auto_scaling_group_name]
+                },
+            ),
+        )
+
+        deploy_rule.add_target(targets.LambdaFunction(assign_eip_lambda))
+        asg_lifecycle_rule.add_target(targets.LambdaFunction(assign_eip_lambda))
+
+    def create_lambda_function(
+        self,
+        eip_allocation_id: secretsmanager.Secret,
+    ) -> lambda_alpha_.PythonFunction:
+        """Create and return the Lambda function."""
+        lambda_role = iam.Role(
+            self,
+            "IalirtEipLambdaRole",
+            assumed_by=iam.ServicePrincipal("lambda.amazonaws.com"),
+            managed_policies=[
+                iam.ManagedPolicy.from_aws_managed_policy_name(
+                    "service-role/AWSLambdaBasicExecutionRole"
+                ),
+                iam.ManagedPolicy.from_aws_managed_policy_name("AmazonEC2FullAccess"),
+                iam.ManagedPolicy.from_aws_managed_policy_name("AutoScalingFullAccess"),
+            ],
+        )
+
+        eip_lambda = lambda_alpha_.PythonFunction(
+            self,
+            id="IalirtAssignEipLambda",
+            function_name="ialirt-eip",
+            entry=str(
+                pathlib.Path(__file__).parent.joinpath("..", "lambda_code").resolve()
```

Comment:
Be consisten in choice of parent traversal. If you are already using parent, might as well use it twice, or put "../.." in the joinpath.
```suggestion
                pathlib.Path(__file__).parent.parent.joinpath("lambda_code").resolve()
```